doc/source/f2py/buildtools/skbuild.rst

```diff
@@ -0,0 +1,84 @@
+.. _f2py-skbuild:
+
+============================
+Using via ``scikit-build``
+============================
+
+``scikit-build`` provides two separate concepts geared towards the users of Python extension modules.
+1. A ``setuptools`` replacement
+2. A series of ``cmake`` modules with definitions which help building Python extensions
+
+.. note::
+
+   It is possible to use ``scikit-build``'s ``cmake`` modules to `bypass the
+   cmake setup mechanism`_ completely, and to write targets which call ``f2py
+   -c``. This usage is **not recommended** since the point of these build system
+   documents are to move away from the internal ``numpy.distutils`` methods.
+
+For situations where no ``setuptools`` replacements are required or wanted (i.e.
+if ``wheels`` are not needed), it is recommended to instead use the vanilla
+``cmake`` setup described in :ref:`f2py-cmake`.
+
+Fibonacci Walkthrough (F77)
+===========================
+
+We will consider the ``fib``  example from :ref:`f2py-getting-started` section.
+
+.. literalinclude:: ./../code/fib1.f
+    :language: fortran
+
+``CMake`` modules only
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Consider using the following ``CMakeLists.txt``.
+
+.. literalinclude:: ./../code/CMakeLists_skbuild.txt
+   :language: cmake
+
+Much of the logic is the same as in :ref:`f2py-cmake`, however notably here the
+appropriate module suffix is generated via ``sysconfig.get_config_var("SO")``.
+The resulting extension can be built and loaded in the standard workflow.
+
+.. code:: bash
+
+    ls .
+    # CMakeLists.txt fib1.f
+    mkdir build && cd build
+    cmake ..
+    make
+    python -c "import numpy as np; import fibby; a = np.zeros(9); fibby.fib(a); print (a)"
+    # [ 0.  1.  1.  2.  3.  5.  8. 13. 21.]
+
+
+``setuptools`` replacement
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The utility of ``scikit-build`` lies in being able to drive the generation of
+more than extension modules, in particular a common usage pattern is the
+generation of Python distributables (for example for PyPI).
+
+The workflow with ``scikit-build`` straightforwardly supports such packaging requirements. Consider augmenting the project with a ``setup.py`` as defined:
+
+.. literalinclude:: ./../code/setup_skbuild.py
+   :language: python
+
+Along with a commiserate ``pyproject.toml``
+
+.. literalinclude:: ./../code/pyproj_skbuild.toml
+   :language: toml
+
+Together these can build the extension using ``cmake`` in tandem with other
+standard ``setuptools`` outputs.
```

Comment:
I'm not sure I understand the second part of this sentence. Why would `setuptools` be involved here?

Reply:
Ah I should reword this, essentially `scikit-build` is used here so the extension module is built with `cmake` driven by `setup.py`. So instead of the standard `mkdir build && cd build && cmake .. && make` one can call `python setup.py build_ext --inplace`.

This makes it easier to integrate in a workflow where multiple extension modules are to be built for example.

Reply:
This is a legacy behavior that scikit-build plans to get rid of, so I would just not mention that it's easier (it's also not really easier, one cannot mix two build systems - at most this is "muscle memory for old-timers").

Reply:
Makes sense, adding a note. Honestly going over this now I'd say that as of CMake 3.17 scikit-build is 100% obsolete but that might be too strong a note, especially since there are many situations where it is easier to add scikit-build than update CMake.

Reply:
I don't think that's quite true. It does too much perhaps and needs some updates, but I don't think you'll find `pyproject.toml` hooks and the ability to build standards-compliant wheels directly in CMake.